src/span.rs

```diff
@@ -0,0 +1,24 @@
+use std::ops::Range;
+
+// A source code span, used for error reporting.
+#[derive(Clone)]
+pub enum Span {
+    // Span is unknown - no source information.
+    Unknown,
+    // Byte range.
+    ByteRange(Range<usize>),
+}
+
+impl Span {
```

Comment:
should this stuff be behind the `span` feature?

Reply:
This would creep the feature flags outside of lib/arena and into all of the lexers. Dead code elimination should take care of it regardless, since without span feature, setting a span does nothing.